### DISCUSSION TITLE
wire:loading elements in Livewire 3 sometimes do not get removed

### DISCUSSION THREAD

**Discussion by Francismori7** [livewire/livewire#6040] [7. Bugs (archived)]
I'm currently migrating a Livewire 2 component set to Livewire 3, knowing it is in beta, of course.
Right now, a page has a wire:loading.delay element that shows when there are any Livewire updates going on.
It works fine (and used to work fine for multiple months), but now with Livewire 3, the element sometimes stays on the page even after the update network request came back and the data has updated in the DOM.
If I manually remove `display: inline-block` from the wire:loading element, the page continues to behave just fine.

Unsure if related, but it shows this in the console:
![image](https://github.com/livewire/livewire/assets/520647/1915c19d-b631-43c2-8487-9f71eb53bb0e)

**Comment by romansh** [comment]
I run into the same yesterday. It does not work with all css display options, like block, inline-block, flex and so on.

**Comment by joshualukecaine** [comment]
I had this issue also where in a blade component  wire:loading was hidding the element on page load, showing it when clicked but then not hiding it again after the round trip

`<div
            role="status"
            wire:loading 
            wire:target='{{ $attributes['wire:click'] }}'          
        >
            <x-ui.spinner 
                wire:key="{{ time() }}-spinner"
                size="xs"
                mode="static"
            />           
        </div> `

A work around that worked was to use a hidden class to begin with and then wire:loading.class.remove="hidden" to remove that class  during the "loading" state

`<div
            role="status"
            class="hidden"
            wire:loading.class.remove="hidden"
            wire:target='{{ $attributes['wire:click'] }}'          
        >
            <x-ui.spinner                 
                size="xs"
                mode="static"
            />           
        </div>`

**Reply by m15studio** [reply]
I have more V2 components with wire:loading
@joshualukecaine solution's works also for me and i'is unique way to have same V2 behaviour on components

**Comment by joshhanley** [comment]
@Francismori7 thanks for reporting! Any chance you could provide a Livewire component class and view that demonstrates the issue with as little code as possible and screenshots so we can investigate? Thanks!

**Reply by Francismori7** [reply]
Unfortunately, by the very nature of its "sometimes happens", it's hard to give a component that would do it. To be fair, it happens a lot when multiple nested components are requesting data.

**Comment by DanielSpravtsev** [comment]
Same there.  

```
<form wire:submit="submit">
<input wire:model="text"/>
<button wire:click="closeModal">
            Close
            <i wire:loading wire:target="closeModal" class="far fa-spinner fa-spin"></i>
</button>
<button type="submit">
            Submit
            <i wire:loading wire:target="submit" class="far fa-spinner fa-spin"></i>
</button>
</form>
```

After submitting form, livewire displays validation error message, but spinner still visible after request ended. No console errors. 
With @joshualukecaine loading.class.remove="hidden" works fine

@joshhanley And one more thing: when there are two buttons, all loading indicators starts running after any click/submit event, without respecting of wire:target

**Comment by joshhanley** [comment]
I have submitted a PR #6233 that fixes the Livewire style priority order which should fix this issue.
